packages/hardhat-core/src/internal/core/params/argumentTypes.ts

```diff
@@ -121,6 +121,50 @@ export const int: CLIArgumentType<number> = {
   },
 };
 
+/**
+ * BigInt type.
+ * Accepts either a decimal string integer or hexadecimal string integer.
+ * @throws HH301
+ */
+export const bigint: CLIArgumentType<bigint> = {
+  name: "bigint",
+  parse: (argName, strValue) => {
+    const decimalPattern = /^\d+(?:n)?$/;
+    const hexPattern = /^0[xX][\dABCDEabcde]+$/;
+
+    if (
+      strValue.match(decimalPattern) === null &&
+      strValue.match(hexPattern) === null
+    ) {
+      throw new HardhatError(ERRORS.ARGUMENTS.INVALID_VALUE_FOR_TYPE, {
+        value: strValue,
+        name: argName,
+        type: bigint.name,
+      });
+    }
+
+    return BigInt(strValue.replace("n", ""));
+  },
+  /**
+   * Check if argument value is of type "bigint".
+   *
+   * @param argName {string} argument's name - used for context in case of error.
+   * @param value {any} argument's value to validate.
+   *
+   * @throws HH301 if value is not of type "bigint"
+   */
+  validate: (argName: string, value: any): void => {
+    const isBigInt = typeof value === "bigint";
```

Comment:
Should we allow inputs of type `number` as long as they're integers and less than `MAX_SAFE_INTEGER`?

Reply:
I suppose not.  That behavior would be consistent with some interfaces, but is not that intuitive and might cause problems.

Reply:
Do we also need a `number` type?  There's a great deal of inconsistency in how numbers are handled across libraries.  I'd like to be able to call tasks from scripts without worrying about what type of number I"m passing in (and for `validate` to check that regular JS numbers are `< MAX_SAFE_INTEGER`).

Reply:
I'm not sure if I understand the `number` type idea.

Reply:
I see. I think I'd personally use `bigint`, but I wouldn't be against adding a new type.